Compilation error using dynamic

Given the following code:

```csharp
dynamic foo = new System.Dynamic.ExpandoObject();
foo.bar="baz";
```

When compiling you will get the following error:

```text
(1,1): error CS0656: Missing compiler required member 'Microsoft.CSharp.RuntimeBinder.CSharpArgumentInfo.Create'
```

Looks like a reference is missing

--- a/src/ScriptCs.Core/ScriptExecutor.cs
+++ b/src/ScriptCs.Core/ScriptExecutor.cs
@@ -21,6 +21,7 @@ namespace ScriptCs
             "System.Xml", 
             "System.Xml.Linq", 
             "System.Net.Http",
+            "Microsoft.CSharp",
             typeof(ScriptExecutor).Assembly.Location,
             typeof(IScriptEnvironment).Assembly.Location
         };
@@ -33,7 +34,8 @@ namespace ScriptCs
             "System.Text", 
             "System.Threading.Tasks",
             "System.IO",
-            "System.Net.Http"
+            "System.Net.Http",
+            "System.Dynamic"
         };
 
         private const string ScriptLibrariesInjected = "ScriptLibrariesInjected";

--- a/src/ScriptCs.Engine.Common/CommonScriptEngine.cs
+++ b/src/ScriptCs.Engine.Common/CommonScriptEngine.cs
@@ -17,9 +17,8 @@ namespace ScriptCs.Engine.Common
         protected CommonScriptEngine(IScriptHostFactory scriptHostFactory, ILogProvider logProvider)
         {
             Guard.AgainstNullArgument("logProvider", logProvider);
-            ScriptOptions = new ScriptOptions().WithReferences(typeof(ScriptExecutor).Assembly, typeof(Object).Assembly);
+            ScriptOptions = new ScriptOptions().WithReferences(typeof(Object).Assembly);
             _scriptHostFactory = scriptHostFactory;
-
             _log = logProvider.ForCurrentType();
         }
 

--- a/test/ScriptCs.Core.Tests/ReplTests.cs
+++ b/test/ScriptCs.Core.Tests/ReplTests.cs
@@ -387,8 +387,8 @@ namespace ScriptCs.Tests
                 _repl.Initialize(Enumerable.Empty<string>(), Enumerable.Empty<IScriptPack>());
                 _repl.Execute("#r \"my.dll\"");
 
-                //default references = 9, + 1 we just added
-                _repl.References.Paths.Count().ShouldEqual(10);
+                //default references = 10, + 1 we just added
+                _repl.References.Paths.Count().ShouldEqual(11);
             }
 
             [Fact]

--- a/test/ScriptCs.Tests.Acceptance/ScriptExecution.cs
+++ b/test/ScriptCs.Tests.Acceptance/ScriptExecution.cs
@@ -63,5 +63,22 @@
             "Then the Env object is displayed"
                 .f(() => output.ShouldContain("ScriptCs.ScriptEnvironment"));
         }
+        
+        [Scenario]
+        public static void ScriptCanUseDynamic(ScenarioDirectory directory, string output )
+        {
+            var scenario = MethodBase.GetCurrentMethod().GetFullName();
+
+            "Given a script which uses dynamic"
+                .f(() => directory = ScenarioDirectory.Create(scenario)
+                    .WriteLine("foo.csx", @"dynamic obj = new ExpandoObject(); obj.foo = ""bar""; Console.WriteLine(obj.foo); ;"));
+
+            "When I execute the script"
+                .f(() => output = ScriptCsExe.Run("foo.csx", directory));
+
+            "Then the dynamic value is properly returned "
+                .f(() => output.ShouldContain("bar"));
+
+        }
     }
 }

